### ISSUE TITLE
Navigating issues

### ISSUE THREAD

**Issue by michael** [substance/texture#678]
We need a way to easily see how many issues there are in a document and some way of navigation. Giuliano suggested not to use another panel (3 column layout) so this is an idea I came up with.

![image](https://user-images.githubusercontent.com/2931/44581079-a6cd2880-a79c-11e8-862c-7aefb64c23ba.png)

Some questions and ideas: 

- Do you support the sketched proposal?
- Do we want a full report with detailed error messages (then maybe we should show the issue list in a full-size popup)
- Do we need to see an aggregated list of issues at all? Maybe it's enough to navigate to "next issue" and see the problem contextually in the main view?

**Comment by michael** [comment]
Just to track Fabio's feedback here:

> Hi Michael,
>
> It is looking great!!
>
> Regarding the warnings, my suggestion is to put a warning icon besides each group within one or more warnings.
> 
> Authors (4) <warning icon>
> Editors (1)
> Groups (1) <warning icon>
> 
> For warning details (if needed) the user can access each group to check its warnings. They could be organized in a collapsible list in the top level of each group.

So we could additionally mark sections that have issues. Just not to add too much symbols and clutter... the collapsible list at the beginning of each section I'd rather avoid.

**Comment by gmaciocci** [comment]
Hi Michael,

I agree about not adding too many warning symbols everywhere and keeping one single access points to a condensed list of all actionable warnings - something that can act as a "work queue" for someone to go through and fix issues.

I just don't know that the top right dropdown is doing. There could be a lot of issues, multiple ones per section, and if they need to be selected one by one by going back to the dropdown each time it could get tedious, as that's a lot of clicks. 

I maintain that the simplest solution is to add a single UI access point which filters the full metadata list to items with issues, while highlighting the relevant fields as you've shown. That way, a staffer could simply click it, use it as their work queue, go through all the issues one by one going down the list without ever switching view.

**Comment by Melissa37** [comment]
I don't totally understand without visuals, but I think I agree with @gmaciocci - thinking about the number of errors that could be encountered once we have all plugged in our various Schematron and other requirements and the number of issues we can see from author files.

**Comment by michael** [comment]
@gmaciocci I think now I understand you... you are suggesting to filter the cards, to those that have `>1` issues? Think that's generally a good idea, and we could avoid returning to the "warnings list". I need to think this through more, since we have a very dynamic nature of our warnings. E.g. one keystroke could turn a card from warning state to ok state, and then ultimately the card would disappear from the filtered view while you are editing one field. We could flag those problematic cards and keep them until you release the filter, but that would somehow be inconsistent, as the cards are no longer problematic.  Just one option to consider: You could cycle through all fields with warnings by using a keyboard shortcut in the primary fully expanded view. For clickers we could have next/previous buttons. Personally I'd see this more as a number of todos, and I'm usually walking through them one by one until they are all fixed. However if you want to get an idea of the quality of the doc, there may be users that want to see a full list of warning messages.

@Melissa37 the validation during editing can't really be compared to schematron verification. Schematron you run on a whole XML file, while our validation is reactive during editing (even works in realtime use-cases). In a first step we will just mark 'required' fields but we can have more complex rules (like value must be a number great than 5 etc.) just like with schematron. If we were able to agree on rules across publishers we could reach a guaranteed quality of the XML output, so that you wouldn't need to run schematron at all because you can already be sure all rules are fulfilled unless there are warnings shown in Texture.

Anyways those are technical details, and there's different ways to approach this. First I think we should have a good UI/UX direction.

**Comment by fabiobatalha** [comment]
@michael 

I did not express myself well. What I was trying to say is something like the image bellow.

![screenshot from 2018-08-24 09-47-30](https://user-images.githubusercontent.com/504142/44588142-f3a40580-a782-11e8-8519-d27a94b0dbcc.png)

A small icon to highlight the groups with errors, and in the central area, a box with all the warnings.

Each group can have its own warning box, this box can be collapsed (maybe a bottom to collapse all warning boxes at the same time)

**Comment by Melissa37** [comment]
I like @fabiobatalha idea, but I think it would be useful to discuss as a group and really dig into the different use-cases.
Depending on the stage of the process, you might let some problems pass, but not others. It's useful to know where the problems are (what type of metadata), at a glance.
For instance, off the top of my head, we might not ask the author to correct some errors in the references, but would do it ourselves, but if we knew there were errors in every reference, and we could see at a glance what the errors are in the image Fabio shared, we'd be able to make a quick judgement.
However, I'd like someone in our production team (@JGilbert-eLife) to validate that thought!

**Comment by fabiobatalha** [comment]
@gmaciocci We do not intend to have different people looking for different types of errors.
It is only a way to organize the errors per group (data type).

In my opinion will be difficult to find a good UX approach to present all errors in only one place.

**Comment by gmaciocci** [comment]
@fabiobatalha I respectfully disagree, I think having a single call to action to only show a list of metadata fields with errors would work. You could further segment that list with a jump menu divided by sections or error types, but  even without it you'd have a single list within which anyone could easily find any erroneous field, and quickly address each error on a case by case basis.
